ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Client.scala

```diff
@@ -361,26 +361,21 @@ private[ember] object H2Client {
     } yield (http1Client: TinyClient[F]) => { (req: Request[F]) =>
       val key = H2Client.RequestKey.fromRequest(req)
       val priorKnowledge = req.attributes.lookup(H2Keys.Http2PriorKnowledge).isDefined
-      val h2Prior = Alternative[Option].guard(priorKnowledge).as(Http2)
-      for {
-        socketType <- Resource.eval(
```

Comment:
Whoa, it seems `socketType` has been computing regardless of the guard's result. This refactoring could also bring saving a few processor ticks. 👏🏻